src/repo/kv.rs

```diff
@@ -101,234 +95,301 @@ impl DataStore for KvDataStore {
     }
 }
 
+// in the transactional parts of the [`Infallible`] is used to signal there is no additional
+// custom error, not that the transaction was infallible in itself.
+
 #[async_trait]
 impl PinStore for KvDataStore {
-    async fn is_pinned(&self, block: &Cid) -> Result<bool, Error> {
-        is_pinned(self, block)
+    async fn is_pinned(&self, cid: &Cid) -> Result<bool, Error> {
```

Comment:
Maybe this is the only place where transaction is not needed, as sled's single-key operations are atomic :)